### ISSUE TITLE
(Qanet verification) 🐞 [Bug]: prevent deploy contracts on Grace period rented node

### ISSUE THREAD

**Issue by A-Harby** [threefoldtech/tfgrid-sdk-ts#3025]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### which package/s did you face the problem with?

Dashboard

### What happened?

After my rented node went to grace period, I tried to deploy a VM on it, and I was able to choose node and have the deploy button active with no error, only an alert saying that the node is on grace period.

And the alert is only shown on manual selection; it doesn't show it on automated selection.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/95569e77-7da5-439c-b6ad-4e58895f782e)

### which network/s did you face the problem on?

QA

### version

2.5.0-rc3

### Relevant screenshots/screen records

It should show an error on node selection with a clear error message.

[screen-capture (36).webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/9038da82-3ed7-4595-82a5-556c8162418b)

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2572

### Relevant log output

```console
.
```

**Comment by samaradel** [comment]
### Work in progress:
Tried to get node on grace period on QA, I didn't get the error that mentioned yet.. still debugging

**Comment by samaradel** [comment]
### Work in progress:
The card with a grace period turned to red, needs to be undetectable too

**Comment by samaradel** [comment]
### Work Completed:
Prevent deployment on nodes with contracts in the grace period

**Comment by Mahmoud-Emad** [comment]
I would recommend excluding these nodes from the filters
also, if this is important info that the user should know, we can display a dialog with the nodes that are excluded. 

But, I think it's not a good behavior to tell the user that this node isn't available after filtering, he user needs to deploy with only a single click.

**Comment by A-Harby** [comment]
Qanet 2.6.0-rc3.

The deployment is not possible for a grace period rented node on the automated Node Selection option.
![image](https://github.com/user-attachments/assets/c1b2b778-ba26-4690-ac53-22eb6faf7fb7)

But it's still the same on manual Node Selection.
![image](https://github.com/user-attachments/assets/e37b9dca-d687-4d8b-b909-b8b133547ead)
The deploy button is active and enabled and gives me this error after clicking it.
![image](https://github.com/user-attachments/assets/d9499217-b37b-40a0-af7f-29244d1dd9d0)

**Comment by samaradel** [comment]
yeah we intended as per https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3191#issuecomment-2267612381

**Comment by A-Harby** [comment]
> yeah we intended as per [#3191 (comment)](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3191#issuecomment-2267612381)

Yes, showing it is correct and the notification is shown as you see in the screenshot, but the deploy button should be disabled as the automated to keep same behavior in both (consistency).

**Comment by samaradel** [comment]
> > yeah we intended as per [#3191 (comment)](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3191#issuecomment-2267612381)
> 
> Yes, showing it is correct and the notification is shown as you see in the screenshot, but the deploy button should be disabled as the automated to keep same behavior in both (consistency).

No, the deploy button is always enabled. When you click on it and you have an invalid input, it will scroll to show it.

**Comment by A-Harby** [comment]
I discussed it with @ramezsaeed, and as per the discussion, we decided it would be better to disable the button, as it's not the best scenario to enable the button click just to show that error.

![image](https://github.com/user-attachments/assets/71d30801-2d7f-4448-b2bd-887f1b0f9d2f)

The notification and disabled button will be good enough.

**Comment by AhmedHanafy725** [comment]
The deploy button shouldn't be disabled in any case, but we can enhance the behavior by checking if the node's rent contract is in the grace period while selecting the node as we check the aliveness and storage pools

**Comment by samaradel** [comment]
### Work Completed:
Check if the rented node contract is in the grace period when listing the node.

**Comment by samaradel** [comment]
I blocked it as I'm waiting for any dedicated node to be available to rent and check.

**Comment by samaradel** [comment]
### Work Completed:
Handle console error

**Comment by samaradel** [comment]
### Work in Progress (WIP):
Investigating the recursive call error

**Comment by samaradel** [comment]
### Work Completed:
Remove toast timeout

**Comment by samaradel** [comment]
### Work Completed: 
Fix recursive calls

**Comment by samaradel** [comment]
### Work Completed:
Remove unwanted contract validation
Early exit for check it node is rented and not in dedicated farm
Remove the check only if the dedicated filter is ON

**Comment by atefnazmy** [comment]
- Verified on Dev,
- Version: 703628d
- The issue has been fixed, and trying to deploy on Grace period rented nodes is not allowed with an alert message declare that "can't deploy on the node because it is in Grace period"

https://github.com/user-attachments/assets/fcd3e837-b231-4967-b37d-05c66c0608f9

#### Related Link:
- [Testcase](https://app.testlodge.com/a/26076/projects/40893/suites/234374?expand_section=422333#case_4260717)
